src/vm/core.zig

```diff
@@ -1096,6 +1138,203 @@ test "trace is disabled" {
     }
 }
 
+// This instruction is used in the functions that test the `deduceOp1` function. Only the
+// `opcode` and `res_logic` fields are usually changed.
+const deduceOp1TestInstr = instructions.Instruction{
+    .off_0 = 1,
+    .off_1 = 2,
+    .off_2 = 3,
+    .dst_reg = .FP,
+    .op_0_reg = .AP,
+    .op_1_addr = .AP,
+    .res_logic = .Add,
+    .pc_update = .Jump,
+    .ap_update = .Regular,
+    .fp_update = .Regular,
+    .opcode = .Call,
+};
+
+test "deduceOp1 when opcode == .Call" {
+    // ************************************************************
+    // *                 SETUP TEST CONTEXT                       *
+    // ************************************************************
+    // Nothing.
+
+    // ************************************************************
+    // *                      TEST BODY                           *
+    // ************************************************************
+    var instr = deduceOp1TestInstr;
+    instr.opcode = .Call;
```

Comment:
If we declare the original instr as `var`, we can do this:

```suggestion
    deduce_op1_test_instr.opcode = .Call;
```

Reply:
Yeah but that creates a race condition if tests execute in parallel